pages/table/sticky-columns.page.tsx

```diff
@@ -0,0 +1,318 @@
+  });
+
+  return (
+    <ScreenshotArea>
+      <h1>Sticky columns</h1>
+      <SpaceBetween size="xl">
+        <Table
+          data-test-id="simple"
+          stickyColumns={{ first: 1, last: 1 }}
+          columnDefinitions={COLUMN_DEFINITIONS}
+          items={ITEMS}
+          sortingDisabled={true}
+          ariaLabels={{
+            tableLabel: 'simple',
+          }}
+          header={<Header>Simple table</Header>}
+        />
+        <Table
+          data-test-id="resizable"
+          resizableColumns={true}
+          stickyColumns={{ first: 1, last: 1 }}
+          columnDefinitions={COLUMN_DEFINITIONS}
+          items={ITEMS}
+          sortingDisabled={true}
+          ariaLabels={{
+            tableLabel: 'Resizable',
+          }}
+          header={<Header>Resizable columns</Header>}
+        />
+        <Table
+          data-test-id="with-collection-preferences"
+          stickyColumns={preferences.stickyColumns}
+          stickyHeader={true}
+          preferences={
+            <CollectionPreferences
+              title="Preferences"
+              confirmLabel="Confirm"
+              cancelLabel="Cancel"
+              onConfirm={({ detail }) => setPreferences(detail)}
+              preferences={preferences}
+              stickyColumnsPreference={{
+                firstColumns: {
+                  title: 'Stick first column(s)',
+                  description: 'Keep the first column(s) visible while horizontally scrolling table content.',
+                  options: [
+                    { label: 'None', value: 0 },
+                    { label: 'First column', value: 1 },
+                    { label: 'First two columns', value: 2 },
+                  ],
+                },
+                lastColumns: {
+                  title: 'Stick last column',
+                  description: 'Keep the last column visible while horizontally scrolling table content.',
+                  options: [
+                    { label: 'None', value: 0 },
+                    { label: 'Last column', value: 1 },
+                  ],
+                },
+              }}
+            />
+          }
+          ariaLabels={{
+            selectionGroupLabel: 'Items selection',
+            allItemsSelectionLabel: ({ selectedItems }) =>
+              `${selectedItems.length} ${selectedItems.length === 1 ? 'item' : 'items'} selected`,
+            itemSelectionLabel: ({ selectedItems }, item) => {
+              const isItemSelected = selectedItems.filter(i => i.name === item.name).length;
+              return `${item.name} is ${isItemSelected ? '' : 'not'} selected`;
+            },
+            tableLabel: 'With collection preferences',
+          }}
+          columnDefinitions={COLUMN_DEFINITIONS}
+          items={ITEMS}
+          sortingDisabled={true}
+          header={<Header>Simple table with collection preferences & sticky-header</Header>}
+        />
+        <Table
+          data-test-id="inline-editing"
```

Comment:
Tabbing between editable cells within a sticky column causes a scroll.

https://user-images.githubusercontent.com/20790937/232777561-d25673f1-9386-4a99-8bbe-3909ed320ded.mov

Reply:
When cell edit mode activates the table scrolls.

https://user-images.githubusercontent.com/20790937/232778018-5d9e6fb4-6316-40e7-ad50-65fa023035d0.mov

Reply:
Similar but weirder scroll effects happen when focusing cells or entering the edit mode that are within the last sticky column. The table scrolls either to the left or to the right.

https://user-images.githubusercontent.com/20790937/232778344-17d28ec8-211f-4729-9ff7-6e72e8c7fd87.mov

Reply:
Same happens for table with selection when tabbing between radio buttons.

Reply:
These issues are less reproducible or not reproducible in Safari.